docker-compose.yml

```diff
@@ -61,6 +61,36 @@ services:
       localnet:
         ipv4_address: 192.168.10.5
 
+  terradnode4:
+    container_name: terradnode4
+    image: "terramoney/terrad-env"
+    environment:
+      - ID=4
+      - LOG=$${LOG:-terrad.log}
+    ports:
+      - "9094:9090"
+      - "26665-26666:26656-26657"
+    volumes:
+      - ./build:/terrad:Z
+    networks:
+      localnet:
+        ipv4_address: 192.168.10.6
+
+  terradnode5:
```

Comment:
it doesn't feel good for me since you cannot scale more node in docker-compose while TESTNET_NVAL is an arg.

Reply:
Oh, yeah. But that was the case before as well. The compose file ALWAYS spins 6 nodes. But the TESTNET_NVAL decides how many of them are validators. If I set TESTNET_NVAL=2, then I have 2 validators and 4 passive full nodes. If I do TESTNET_NVAL=12, then I get 6 validators.

Reply:
The problem is, that you cannot scale the compose file - except you generate the compose file beforehand.

Reply:
We can make it work but it will requires some rework and abandoning the static IP configuration. As such all container names in  docker-compose are deterministic and they also resolve via DNS if running  in a user-defined network (!= host). So we are able to rebuild the docker tooling to rely on the container names (since they are resolved to the IP automagically by docker) and thus be able to scale above whatever number is statically configured. It will however take some time to do this rework which is why I have been holding off on it so we can focus on our primary issues until we are past the two network upgrades.

If it helps we can schedule a session on docker DNS and do a deep dive on the subject or you can just leave it for me to find time to sort it out . :)

Reply:
Is this resolved?

Reply:
Yes :)